### PR TITLE
ED-1759 upload logo

### PR DESCRIPTION
**Pull request by kowalcj0**
This implements 2 scenarios listed below:
```gherkin
    @ED-1759
    @profile
    @logo
    Scenario Outline: Supplier should be able to upload an image to set company’s logo
      Given "Peter Alder" has created and verified profile for randomly selected company "Y"

      When "Peter Alder" uploads "<valid_image>" as company's logo

      Then "Peter Alder" should see that logo on FAB Company's Directory Profile page
      And "Peter Alder" should see that logo on FAS Company's Directory Profile page

      Examples:
        | valid_image              |
        | Anfiteatro_El_Jem.jpeg   |
        | Kobe_Port_Tower.jpg      |
        | Wikipedia-logo-v2-en.png |

    @ED-1759
    @profile
    @logo
    Scenario Outline: Supplier should be able to replace an existing company's logo with a new one
      Given "Peter Alder" has created and verified profile for randomly selected company "Y"
      And "Peter Alder" has set "<original>" picture as company's logo
      And "Peter Alder" can see that logo on FAB Company's Directory Profile page

      When "Peter Alder" uploads "<new_picture>" as company's logo

      Then "Peter Alder" should see that logo on FAB Company's Directory Profile page
      And "Peter Alder" should see that logo on FAS Company's Directory Profile page

      Examples:
        | original               | new_picture         |
        | Anfiteatro_El_Jem.jpeg | Kobe_Port_Tower.jpg |
```

All functional tests pass when ran locally against `DEV`:
```
(.venv) [ED-1759-upload-logo]jk@w:directory-tests 10:42:31 $ behave -k --format progress3 --no-logcapture --tags=-wip --tags=-skip tests/functional/features
Supplied path: "tests/functional/features"
Trying base directory: /home/jk/git/dit/directory-tests/tests/functional/features
Behave log file: ./tests/functional/reports/behave.log
Trade Profile    # tests/functional/features/fab/profile.feature
  Supplier should receive a verification email after successful registration - export status is "Yes, in the last year" -- @1.1   .......
  Supplier should receive a verification email after successful registration - export status is "Yes, 1 to 2 years ago" -- @1.2   .......
  Supplier should receive a verification email after successful registration - export status is "Yes, but more than 2 years ago" -- @1.3   .......
  Supplier should receive a verification email after successful registration - export status is "No, but we are preparing to" -- @1.4   .......
  Unauthenticated Suppliers should be able to verify their email address via confirmation link sent in an email  ......
  Suppliers without verified email should be told to verify the email address first before being able to log in  ....
  Supplier should be able to build the Directory Profile once the email address is confirmed  .........
  Supplier should be able to verify company using code sent in the verification letter  ........
  Once verified Company's Directory Profile should be published on FAS  ...
  Some suppliers are not appropriate to feature in the Find a Buyer service  .....
  Suppliers with unverified company profile should be able to logout and log back in  .......
  Suppliers with a standalone SSO/great.gov.uk account should be able to select their company for Directory Profile creation -- @1.1   .......
  Suppliers with a standalone SSO/great.gov.uk account should be able to select their company for Directory Profile creation -- @1.2   .......
  Suppliers with a standalone SSO/great.gov.uk account should be able to select their company for Directory Profile creation -- @1.3   .......
  Suppliers with a standalone SSO/great.gov.uk account should be able to select their company for Directory Profile creation -- @1.4   .......
  Suppliers with a standalone SSO/great.gov.uk account should be able to create a Directory profile  .........
  Suppliers with verified company profile should be able to logout and log back in  .....
  Supplier should be able to upload an image to set company’s logo -- @1.1   ....
  Supplier should be able to upload an image to set company’s logo -- @1.2   ....
  Supplier should be able to upload an image to set company’s logo -- @1.3   ....
  Supplier should be able to replace an existing company's logo with a new one -- @1.1   ......

SSO profile    # tests/functional/features/sso/profile.feature
  Suppliers should receive an email verification msg after creating a standalone SSO/great.gov.uk account  ....
  Suppliers should be able to confirm email address for a standalone SSO/great.gov.uk account  .....

2 features passed, 0 failed, 0 skipped
23 scenarios passed, 0 failed, 9 skipped
139 steps passed, 0 failed, 27 skipped, 0 undefined
Took 12m13.595s
```

### CODE REVIEW

File: tests/functional/features/context_utils.py [hunk 1]
```diff
@@ -107,6 +108,16 @@ def set_company_for_actor(self, actor_alias, company_alias):
         logging.debug("Could not find an actor aliased '%s'", actor_alias)
 
 
+def set_company_logo_detail(self, alias, *, picture=None, url=None, hash=None):
```
Comment (sdonk) [inline tests/functional/features/context_utils.py]: Why the `*`? Are we expecting other arguments?

File: tests/functional/features/steps/fab_when_impl.py [hunk 2]
```diff
@@ -1184,3 +1190,89 @@ def sso_go_to_create_trade_profile(context, supplier_alias):
...
+    company = context.get_unregistered_company(actor.company_alias)
+    filename = get_absolute_path_of_file(picture)
+
+    # Upload company's logo
+    headers = {"Referer": get_absolute_url("ui-buyer:upload-logo")}
+    url = get_absolute_url("ui-buyer:upload-logo")
+    data = {"csrfmiddlewaretoken": actor.csrfmiddlewaretoken,
+            "supplier_company_profile_logo_edit_view-current_step": "logo",
+            }
+    files = {"logo-logo": open(filename, "rb")}
```
Comment (sdonk) [inline tests/functional/features/steps/fab_when_impl.py]: It looks like this file object will never get closed

File: tests/functional/features/context_utils.py (hunk 1 above)
Comment (kowalcj0) [inline tests/functional/features/context_utils.py]: All the arguments that follow the `*` are optional and this gives us more flexibility when you need to update the specific property of company's logo.

File: tests/functional/features/steps/fab_when_impl.py (hunk 2 above)
Comment (kowalcj0) [inline tests/functional/features/steps/fab_when_impl.py]: ouch, true! will look into this.

File: tests/functional/features/context_utils.py (hunk 1 above)
Comment (kowalcj0) [inline tests/functional/features/context_utils.py]: it's better explained here http://www.asmeurer.com/python3-presentation/slides.html#9

**Pull request merged**